torch/csrc/inductor/aoti_runtime/model.h

```diff
@@ -105,9 +105,16 @@ class AOTInductorModelBase {
     parse_device_str(device_str, device_type_, device_idx_);
 
 #ifdef USE_CUDA
+    AOTI_RUNTIME_CHECK(
+        device_type_ == aoti_torch_device_type_cuda(),
+        "Can't instantiate CUDA model for " + device_str);
     if (device_idx_ == -1) {
       AOTI_RUNTIME_DEVICE_CHECK(cudaGetDevice(&device_idx_));
     }
+#else
+    AOTI_RUNTIME_CHECK(
+        device_type_ == aoti_torch_device_type_cpu(),
+        "Can't instantiate CPU model for " + device_str);
```

Comment:
why can I not instantiate a CPU model on a CUDA build?  Then how do I build an aoti runner for cpu on a host that has cuda installed?  

I do presume all the operators exist?  And the fact that we prepare tensors on cpu and then ship to cuda mean we can instantuiate cpu tensors..?

Reply:
I expected that for cpu dso loaded on a rubber for AOTI, all we’d have to do is change the kDevice that is used to move input values to the proper device for processing. If it’s a cpu model we can just set that to cpu and make a null move?

Reply:
I have been chatting to @malfet offline on this. Here, we are relying on the `USE_CUDA` macro for checking, and this macro is specified when building model.so, basically an indication of whether this model.so is for CPU or GPU. In the ideal case, we should be able to run a CPU model.so with AOTIModelContainerRunnerCuda, but not the other way around. Initially I was ok with having another follow-up PR to support such a more general case, but looks like this PR already hit that in the existing unit test, `test_pytree_inputs`, although it's probably not intentional (likely that test forgot to specify tensor device).